lib/tasks/plugins.rake

```diff
@@ -25,6 +25,28 @@ namespace :plugin do
       ActiveRecord::Tasks::DatabaseTasks.migrations_paths << path
     end
   end
+
+  task :rubocop, [:engine, :junit] => :environment do |t, args|
+    unless args[:engine]
+      abort("You must specify the name of the plugin (e.g. rake plugin:rubocop['my_plugin'])")
+    end
+
+    engine = "#{args[:engine].tr('-', '_').camelize}::Engine".constantize
+    options = []
+
+    options += ['--format', 'progress']
+    options += ['--format', 'junit', '--out', args[:junit]] if args[:junit]
+
+    config_path = engine.root.join('.rubocop.yml')
+    options += ['--config', config_path.to_s] if config_path.exist?
+
+    options << engine.root.join('{app,lib,test}/**/*.rb').to_s
+
+    require 'rubocop'
```

Comment:
I believe this is the main issue here: This requires the rubocop version set in core and ignores the version specified in the plugin's gemfile.
As a plugin author you want to be able to pin a specific rubocop versions because new versions will include new cops and causing your ci to fail. It would be better to upgrade the version and introduce and fix new cops in a dedicated PR that is tested and then merged.